frontend/public/components/routes/create-route.tsx

```diff
@@ -196,11 +227,54 @@ export class CreateRoute extends React.Component<null, CreateRouteState> {
       }));
   }
 
+  newAltServiceEntry() {
+    return {
+      entry: {
+        service: null,
+        weight: 100,
+      },
+      uid: _.uniqueId(),
+    };
+  }
+
+  addAltServiceEntry(){
+    this.setState({
+      alternateServices: _.concat(this.state.alternateServices, this.newAltServiceEntry()),
+    });
+  }
+
+  removeAltServiceEntry(alternateServiceID){
+    const updatedServiceEntriesArray = [...this.state.alternateServices];
+    updatedServiceEntriesArray.splice(alternateServiceID, 1);
+    this.setState({
+      alternateServices: updatedServiceEntriesArray,
+    });
+  }
+
+  onDataChanged(updatedEntry, entryID) {
```

Comment:
This also needs to call `setState` with a function

Reply:
Call it `index` instead of `entryID` since that's what it is. It will also avoid confusion between `uid` and `entryID`, which are not the same value.